go/vt/vtgate/planbuilder/route.go

```diff
@@ -654,12 +647,21 @@ func (rb *route) UnionCanMerge(right *route) error {
 		}
 		return errIntermixingUnsupported
 	}
+	return rb.isSameShardedRoute(right)
+}
 
+// isSameShardedRoute returns nil if the supplied route has
+// the same single shard target as the current route. If not, it
+// returns an appropriate error.
+func (rb *route) isSameShardedRoute(right *route) error {
 	if rb.ERoute.Opcode != engine.SelectEqualUnique || right.ERoute.Opcode != engine.SelectEqualUnique {
-		return errors.New("unsupported: UNION on multi-shard queries")
+		return errors.New("unsupported: UNION or subquery on multiple shards")
```

Comment:
I actually feel like the previous error message was clearer since it _is_ ok to UNION between two single-shard queries.

How about "unsupported: UNION or subquery containing multi-shard queries"